algebird-core/src/main/scala/com/twitter/algebird/Ring.scala

```diff
@@ -151,16 +151,16 @@ object FloatRing extends Ring[Float] {
     else
       Some {
         val iter = t.toIterator
-        var sum = iter.next().toDouble
-        var c = 0.0
```

Comment:
Why not use Double here? CPUs aren't any faster at float than they are double for basic operations are they? Can't it in principle avoid error when you have overflow in floats that wouldn't overflow a Double?

Also note: the code did this originally, so this would be a change from the current branch not just my PR.

Reply:
Oh, I just did the change to see if it mattered - I was more tossing this in to accompany my question to you about if / why it mattered :) that is a good call too about avoiding overflow in principle. I’ll revert this now so you can merge cleanly